app/services/experiment_control_service.rb

```diff
@@ -0,0 +1,94 @@
+class InvalidExperiment < StandardError; end
+
+class ExperimentControlService
+  class << self
+    def start_current_patient_experiment(name, days_til_start, days_til_end)
+      raise ArgumentError, "Start date must be before end date" if days_til_end < days_til_start
+
+      existing_experiment = Experimentation::Experiment.where(experiment_type: "current_patient_reminder", state: ["selecting", "live"])
+      raise InvalidExperiment, "A current patient experiment is already in progress" if existing_experiment.any?
```

Comment:
This could also be enforced on the model...and probably would be good to do for added safety.

Reply:
You're making good points about model validations. As I said in the other PR, I was planning to write a script for adding experiment data soon and intended to add more validations at that time, but it looks like some of it is necessary here.